test/functional/api/proc_spec.lua

```diff
@@ -62,7 +62,7 @@ describe('API', function()
     it('returns process info', function()
       local pid = funcs.getpid()
       local pinfo = request('nvim_get_proc', pid)
-      eq((iswin() and 'nvim.exe' or 'nvim'), pinfo.name)
+      eq((is.os.win and 'nvim.exe' or 'nvim'), pinfo.name)
```

Comment:
This is too clever and too nested. We need to stop this "builder pattern"-like cleverness. `is_os` and `is_ci` are not really related except in there shared "verb".

Reply:
Whare are your thoughts on using two tables like @lewis6991 suggested instead of functions?

Reply:
If you're referring to this:

> I'd personally prefer these weren't functions and are just calculated upfront when helpers.lua is required.

The uname result at least was calculated/cached.